types/dec_coin.go

```diff
@@ -265,14 +376,22 @@ func (coins DecCoins) AmountOf(denom string) Dec {
 	}
 }
 
-// has a negative DecCoin amount
-func (coins DecCoins) HasNegative() bool {
-	for _, coin := range coins {
-		if coin.Amount.IsNegative() {
-			return true
+// IsEqual returns true if the two sets of DecCoins have the same value.
+func (coins DecCoins) IsEqual(coinsB DecCoins) bool {
+	if len(coins) != len(coinsB) {
+		return false
+	}
+
+	coins = coins.Sort()
+	coinsB = coinsB.Sort()
+
+	for i := 0; i < len(coins); i++ {
+		if coins[i].Denom != coinsB[i].Denom || !coins[i].Amount.Equal(coinsB[i].Amount) {
```

Comment:
Can we use the `.Equals` function on the `Coin` type here?

Reply:
Yes, we can. Updated.